vhd/packer/vhd-image-builder-20.04.json

```diff
@@ -0,0 +1,200 @@
+{
```

Comment:
I did the diff to see what differences were needed for this.  Was happy to see it was almost nothing but sad to see it required a whole copy of the file.

This is a quick way to make a new one but there are a total of 3 lines that are different, and only of them is a templated line to begin with.  (Unfortunately, a change in case)

For this quick hack on end of life code, this may be fine.  It is interesting that this is not just a templated update of the vhd-image-builder.json



```
$ diff -u vhd-image-builder.json vhd-image-builder-20.04.json 
--- vhd-image-builder.json      2023-01-06 13:05:01.746380386 +0000
+++ vhd-image-builder-20.04.json        2023-01-06 13:05:01.746380386 +0000
@@ -26,8 +26,8 @@
             "os_type": "Linux",
             "os_disk_size_gb": 30,
             "image_publisher": "Canonical",
-            "image_offer": "UbuntuServer",
-            "image_sku": "{{user `ubuntu_sku`}}-LTS",
+            "image_offer": "0001-com-ubuntu-server-focal",
+            "image_sku": "20_04-lts",
             "image_version": "latest",
             "azure_tags": {
                 "os": "Linux",
@@ -186,7 +186,7 @@
             "type": "file",
             "direction": "download",
             "source": "/opt/azure/vhd-install.complete",
-            "destination": "release-notes-18.04.txt"
+            "destination": "release-notes-20.04.txt"
         },
         {
             "type": "shell",
```

Reply:
I agree.
For Windows I checked in some variables files for the different Windows versions we support(ed) (https://github.com/Azure/aks-engine/blob/master/vhd/packer/windows-2019-vars.json and https://github.com/Azure/aks-engine/blob/master/vhd/packer/windows-2004-vars.json) and just updated the main package job definition to use those (ex: https://github.com/Azure/aks-engine/blob/master/vhd/packer/windows-vhd-builder.json#L59)

Here is where our CI passes in the variable files - https://github.com/Azure/aks-engine/blob/c8e3e1e3dcacd7f326457be4ce59fba1c4c5b87d/packer.mk#L7-L8

Doing something similar shouldn't be too much work and would be much easier to maintain.

Reply:
Good observations. Given the urgency of 18.04-LTS and the minimal long-term maintenance needs (6-9 months) I think there is value in doing zero add'l work given that this current, non-templated implementation works as-is.

My 2¢

Reply:
Fair enough. This OK as is.